### PR TITLE
Make layout explorer the default tab.

### PR DESCRIPTION
**Pull request by jacob314**
Fix some lifecycle bugs in the inspector and layout explorer code so this is ok.
Fixed a lifecycle bug in the service manager as well.

### CODE REVIEW

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart [hunk 1]
```diff
@@ -224,7 +208,8 @@ class InspectorController extends DisposableController
 
   bool programaticSelectionChangeInProgress = false;
 
-  InspectorTreeNode selectedNode;
+  ValueListenable<InspectorTreeNode> get selectedNode => _selectedNode;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/inspector_controller.dart:211]: use ValueListenable pattern instead so we can auto dispose.

File: packages/devtools_app/lib/src/inspector/inspector_tree_flutter.dart [hunk 2]
```diff
@@ -77,17 +77,20 @@ class _InspectorTreeRowState extends State<_InspectorTreeRowWidget>
...
-  InspectorControllerClient get client => _client;
-  InspectorControllerClient _client;
+  final Set<InspectorControllerClient> _clients = {};
 
-  set client(InspectorControllerClient value) {
-    if (_client == value) return;
-    // Do not set a new client if there is still an old client.
-    assert(value == null || _client == null);
-    _client = value;
+  void addClient(InspectorControllerClient value) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/inspector_tree_flutter.dart:82]: due to order disposed methods were called the previous checks in this method were too strict and caused issues.

File: packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart [hunk 3]
```diff
@@ -30,9 +32,11 @@ class _LayoutExplorerTabState extends State<LayoutExplorerTab>
     if (BoxLayoutExplorerWidget.shouldDisplay(node)) {
       return BoxLayoutExplorerWidget(controller);
     }
-    return const Center(
+    return Center(
       child: Text(
-        'Currently, Layout Explorer only supports Box and Flex-based widgets.',
+        node != null
+            ? 'Currently, Layout Explorer only supports Box and Flex-based widgets.'
+            : 'Select a widget to view its layout.',
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart:39]: We could also leave the layout explorer window completely blank when no widget is selected. @InMatrix @cobblest any thoughts?

File: packages/devtools_app/lib/src/service_manager.dart [hunk 4]
```diff
@@ -755,21 +755,15 @@ class ServiceExtensionManager extends Disposer {
...
           return;
         }
-        for (String extension in isolate.extensionRPCs) {
-          await _maybeAddServiceExtension(extension);
-          if (isolateRef != _mainIsolate.value) {
-            // Isolate has changed again.
-            return;
-          }
-        }
+        await Future.wait([
```
Comment (jacob314) [inline packages/devtools_app/lib/src/service_manager.dart:758]: Without this we had super confusing bugs as our app state would reach an async gap with only some of the service extensions added. Now they are all added at the same block.

**Comment by jacob314** [comment]
Fixes https://github.com/flutter/devtools/issues/2642

File: packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart:39]: Or should we always select the root if no widget is selected?

Comment (InMatrix) [reply packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart:39]: Leave the layout explorer window there could help discovery. As an enhancement to the current message, can we suggest a widget to inspect? For example, the second line can be modified to 'Try it out: select the first `Row` widget to view its layout.`  [Go button].

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart (hunk 3 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart:39]: I like the suggestion of finding the first Row widget. Just need to know what alternate message to show if there are no Row/Column widgets in the app.

Comment (InMatrix) [reply packages/devtools_app/lib/src/inspector/layout_explorer/layout_explorer.dart:39]: How about linking to online documentation in that case? We could also consider embedding a short video demo in that space.

**Pull request merged**